orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,218 @@
+from collections import Counter
+from typing import Optional
+
+from AnyQt.QtCore import Qt, QAbstractTableModel, QSize
+from AnyQt.QtWidgets import QSizePolicy, QApplication, QTableView
+from AnyQt.QtGui import QBrush, QColor
+
+
+from Orange.data import Table
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.preprocess import Preprocessor, WordPunctTokenizer
+
+
+class IO:
+    DATA = 'Corpus'
+
+
+class ConcordanceModel(QAbstractTableModel):
+    "A model for constructing concordances from text."
+    def __init__(self):
+        QAbstractTableModel.__init__(self)
+        self.word = None
+        self.data = None
+        self.indices = None
+        self.doc_count = None
+
+    def set_word(self, word):
+        self.modelAboutToBeReset.emit()
+        self.word = word
+        self._compute_indices()
+        self.modelReset.emit()
+
+    def set_data(self, data):
+        self.modelAboutToBeReset.emit()
+        self.data = data
+        self._compute_indices()
+        self.modelReset.emit()
+
+    def flags(self, _):
+        return Qt.ItemIsEnabled | Qt.ItemIsSelectable
+
+    def rowCount(self, parent=None, *args, **kwargs):
+        return len(self.doc_count) if self.doc_count is not None else 0
+        # return len(self.indices) if self.indices is not None else 0
+
+    def columnCount(self, parent=None, *args, **kwargs):
+        return 3
+
+    def data(self, index, role=Qt.DisplayRole):
+        row, col = index.row(), index.column()
+        width = 5
+
+        if role == Qt.DisplayRole:
+            doc = row
+            beg = ''
+            mid = ''
+            end = ''
+            for i_doc, index in self.indices:
+                if i_doc == doc:
+                    beg += ' '.join(self.data.tokens[doc][index-width:index]) + '\n'
+                    mid += self.data.tokens[doc][index] + '\n'
+                    end += ' '.join(self.data.tokens[doc][index+1:index+width]) + '\n'
```

Comment:
I overlooked that you have `+=`, not `=`. But now the text doesn't fit the column at all, does it?